samza-yarn/src/main/java/org/apache/samza/job/yarn/YarnContainerRunner.java

```diff
@@ -110,6 +111,7 @@ public void runContainer(String samzaContainerId, Container container, CommandBu
     log.info("Container ID {} using command {}", samzaContainerId, command);
 
     Map<String, String> env = getEscapedEnvironmentVariablesMap(cmdBuilder);
+    env.put(ShellCommandConfig.ENV_EXECUTION_ENV_CONTAINER_ID(), Util.envVarEscape(container.getId().toString()));
```

Comment:
You should probably add this environmental variable to ShellCommandBuilder and not here?? Please check how other environmental variables are being passed to the containers.